pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -639,7 +639,16 @@ func (c *Controller) onNodeEvent(obj interface{}, event model.Event) error {
 			}
 		}
 		if k8sNode.address == "" {
-			return nil
+			// use internal address if available
+			for _, address := range node.Status.Addresses {
+				if address.Type == v1.NodeInternalIP && address.Address != "" {
```

Comment:
Can you explain why _internal_ address makes sense? We are trying to access this node externally, so I would expect internal to not work?

Reply:
Just an alt for no External IP nodes, node's internalIP always used for "out of cluster" communication under the **private cloud** architecture.

Reply:
Per https://kubernetes.io/docs/concepts/architecture/nodes/#addresses, this field should be for _reaching it within the cluster_. In this case, we explicitly want to reach it from OUT of the cluster, so this feels wrong to me.